woocommerce/woocommerce-ios
# [Performance] Remove unused Leaderboards support

Closes: #10096


## Description

Now that we've replaced the request to the `/wc-analytics/leaderboards` endpoint with a request to `/wc-analytics/reports/products` for the Top Performers data on My Store, we no longer use leaderboards data in the app. This PR removes the unused leaderboards support.

### Changes

* Removes `LeaderboardsRemote`.
* Removes leaderboard Networking models: `Leaderboard`, `LeaderboardRow`, `LeaderboardRowContent`.
* Removes the mapper `LeaderboardListMapper`.
* Removes the Yosemite helper `LeaderboardStatsConverter`.
* Removes unneeded mock API responses.

## Testing instructions


As of #10113 leaderboards aren't being used in the app, but you can confirm they aren't needed with these steps:

1. Build and run the app.
2. Confirm the Top Performers list loads on the My Store tab, and shows the expected data.

---
